### ISSUE TITLE
[FEATURE REQUEST] JSON parsing of cmd.run output.

### ISSUE THREAD

**Issue by ebusto** [saltstack/salt#61784]
Many commands, such as `lldpcli`, can emit output in JSON and other formats. Currently `cmd.run` and its relatives return the output as a string, but it would be nice if the decoding could be done by Salt before the output is returned. It is much easier to visually parse well structured output than it is a wall of text.

This would be very similar to what is provided by `salt.pillar.cmd_json`, except for ad-hoc command execution.

A trivial implementation:
```
ebusto@mana nvcmd % cat __init__.py 
import salt.utils.json


def __virtual__():
    return True


def run_json(cmd):
    """
    Execute the passed command and return the JSON encoded output.
    """

    return salt.utils.json.loads(__salt__['cmd.run'](cmd))
```

**Comment by OrangeDog** [comment]
I'm not sure how useful it would be.

If you're writing custom stuff that uses it, then you can just parse the string yourself (as above). If you're on the CLI you can e.g.:
```
salt-call cmd.run lldpcli --out=json | jq '.local|fromjson'
```
```
salt-call cmd.run lldpcli | tail -n+2 | jq
```

**Comment by ebusto** [comment]
One way to look at this is bringing parity between `cmd.run` and the countless core modules that return structured data. Piping the output through `jq` only ensures the JSON string is prettier; it is still a string.

Custom code that uses the Salt API requires an awkward two phase process of first decoding it as a string, and then as JSON.

**Comment by OrangeDog** [comment]
Custom code has easy access to a JSON parser. Most `cmd.run` returns are not JSON. There are probably more core modules that don't return structured data than those that do. I don't see how or why there should be parity.

**Comment by whytewolf** [comment]
I agree with @OrangeDog this isn't the job of cmd.run. in fact something like this adds a layer of complexity and a fail point to what is one of the more robust functions.

**Comment by tacerus** [comment]
It would be useful to construct variables in templates.

**Comment by whytewolf** [comment]
> It would be useful to construct variables in templates.

`load_json` already exists. and can load any string that is json formatted into a python object. 

```
root@salt00:/srv/salt/tests# salt-call slsutil.renderer salt://tests/tests.sls default_renderer=jinja
local:

    bar
root@salt00:/srv/salt/tests# cat tests.sls
{% set var = salt["cmd.run"]("echo {\\\"foo\\\": \\\"bar\\\"}")|load_json%}
{{var["foo"]}}
```

**Comment by tacerus** [comment]
Thank you, @whytewolf - that's very useful.
